modules/ding_place2book/ding_place2book.admin.inc

```diff
@@ -0,0 +1,376 @@
+<?php
+/**
+ * @file
+ * Contains all administration forms and related functions.
+ */
+
+/**
+ * Form handler for configuration service settings.
+ */
+function ding_place2book_service_admin_form($form, &$form_state) {
+  $form['#tree'] = TRUE;
+  // Gets stored configs.
+  $path = drupal_get_path('module', 'ding_place2book');
+  $path .= '/lib/p2b/p2b.conf';
+  $data = file_get_contents($path, FILE_USE_INCLUDE_PATH);
```

Comment:
what's the point in reading & writing configuration to a file, rather than using variable_get/variable_set? - also, it'll get reset whenever the code is updated.